cmd/krew/cmd/upgrade.go

```diff
@@ -104,7 +105,9 @@ kubectl krew upgrade foo bar"`,
 					return errors.Wrapf(err, "failed to upgrade plugin %q", pluginDisplayName)
 				}
 				fmt.Fprintf(os.Stderr, "Upgraded plugin: %s\n", pluginDisplayName)
-				internal.PrintSecurityNotice(plugin.Name)
+				if indexName == constants.DefaultIndexName {
```

Comment:
TestKrewInstallDoesntShowSecurityWarningForCustomIndex

Reply:
+1
Even if it looks redundant, it is an important documentation of our intended logic.